optuna/integration/sklearn.py

```diff
@@ -29,14 +36,14 @@
     _import_error = e
     _available = False
 
-from optuna import distributions
-from optuna import exceptions
-from optuna import logging
+if _available:
+    if _sklearn_version >= '0.22.1':
+        from sklearn.utils.validation import _check_fit_params as _sklearn_check_fit_params
+    else:
+        from sklearn.utils.validation import _index_param_value as _sklearn_index_param_value
+
 from optuna import samplers  # NOQA
-from optuna import structs
-from optuna import study as study_module
 from optuna import trial as trial_module  # NOQA
-from optuna import type_checking
```

Comment:
A minor but let's be consistent with placing `optuna` import after the Python standard library, i.e. keeping them where they were. You could add `  #NOQA` to these imports.

Reply:
@hvy Sorry for breaking a consistency, fixed in 4023962.